### ISSUE TITLE
Bot isn't claiming characters from namelist

### ISSUE THREAD

**Issue by jodersi** [vivinano/MudaeAutoBot#182]
It's strange that this is happening because it was working before, and while I have added characters to the name list since then as well as removing the characters which have been claimed, I haven't changed the convention to which they're listed. More specifically, the bot doesn't even recognize that something from the name list was pulled in the first place, it just gives the typical someone else rolled in channel ID response, and the characters I have on my name list have all not been claimed. Any idea as to what went wrong?

**Comment by jodersi** [comment]
Update, it started working again when I added a couple more characters and turned it back on just to see if it would work, and then stopped working again when I did the same thing later. I feel like it has to be something with the way I'm entering in the characters, but I can't for the life of me think of how I can be entering it in wrong since it's always been in the same format and just seems to work or not arbitrarily.

**Comment by vivinano** [comment]
It depends if you have already claimed this seems like you might not have had a claim available hence it will stop searching until the next claim cycle this is added to prevent your user from looking like to much of a bot

**Comment by jodersi** [comment]
> It depends if you have already claimed this seems like you might not have had a claim available hence it will stop searching until the next claim cycle this is added to prevent your user from looking like to much of a bot

In the instances where it happened I still had a claim, is there anything I could have done to make the bot think I had claimed when I hadn't?

**Comment by vivinano** [comment]
It is triggered on a update_message so if a message is updated by mudae that fit the criteria then maybe but other than that I haven't heard of any false triggers 

If you can recreate it then I can fix otherwise can't fix what I can't reproduce

**Comment by vivinano** [comment]
So can't reproduce this issue can you provide screenshots of an "active" claim and a missed character , name in the settings , and a screenshot of the logs

**Comment by jodersi** [comment]
Sorry, forgot to send the comment I typed out a couple days ago. It happens when I add a name to the name list in the middle of the list instead of adding it to the end.

**Comment by vivinano** [comment]
And you have validated it with the json validator

**Comment by vivinano** [comment]
So what I assume if your json is not validated then when it goes thru your list only the 1st character will be claimed as it stops when it hits the failure name

**Comment by jodersi** [comment]
ok, how do I validate it? can I use any validation tool online or is there something you recommend?

**Comment by vivinano** [comment]
> also will I have to validate it every time I edit the file?

No I don't unless your very bad at json format then I would recommend it Of course

**Comment by jodersi** [comment]
the validator says that it's valid

**Comment by jodersi** [comment]
although it did change the format by making the list vertical instead of being in a single line

**Comment by vivinano** [comment]
Alright so now also keep in mind that name is exact match only and take a screenshot of what I listed above

**Comment by jodersi** [comment]
I went through and checked every name, they are all exact matches even down to capitalization

**Comment by vivinano** [comment]
And you positive your in a claim cycle and the issue is that is only take the 1st character and not the one in the middle ?

**Comment by jodersi** [comment]
I'm positive that I had a claim when the characters popped up, but its not that it only takes the 1st character its that it doesn't recognize that a character from the name list has been rolled when they roll

**Comment by vivinano** [comment]
Figure out why the 1st character is being claimed it should state what triggers get triggered

**Comment by jodersi** [comment]
It's working now, maybe there was something wrong with the jsons format I didn't recognize that was fixed by the validator/formatting tool.

**Comment by vivinano** [comment]
Maybe the curly brackets unsure but glad it is fixed

**Comment by jodersi** [comment]
> Maybe the curly brackets unsure but glad it is fixed

Ok, its been a while, but 3 months later I've realized what was wrong after a specific instance I encountered earlier today, and I went back to all the other cases of it not claiming in the past and it all lines up. I've realized that the problem that I had every time the bot failed to claim something is that the bot is refreshing the claim window an hour before the server refreshes the claim window. So our server refreshes claim every 3 hours at 3:00, 6:00, and 9:00, the bot is refreshing its claim window at 2:00, 6:00, and 8:00. This has been making it so that if I claim something in the final rolls of a claim period, the bot won't work until the last hour of the next claim period. How can I fix this so that the bot is refreshing its claim period at the same time as the server?
